wren-ai-service/src/providers/__init__.py

```diff
@@ -70,12 +70,15 @@ def llm_processor(entry: dict) -> dict:
     """
     others = {k: v for k, v in entry.items() if k not in ["type", "provider", "models"]}
     returned = {}
-    for model in entry["models"]:
-        model_name = f"{entry['provider']}.{model['model']}"
+    for model in entry.get("models", []):
+        model_name = f"{entry.get('provider')}.{model.get('model')}"
         returned[model_name] = {
-            "provider": entry["provider"],
-            "model": model["model"],
-            "kwargs": model["kwargs"],
+            "provider": entry.get("provider"),
+            "model": model.get("model"),
+            "kwargs": model.get("kwargs"),
+            "api_base": model.get("api_base"),
+            "api_version": model.get("api_version"),
+            "api_key_name": model.get("api_key_name"),
```

Comment:
```python
addition = {k: v for k, v in model.items() if k not in ["model", "kwargs"]}
```

perhaps we could use an additional dict to store the optional items defined at the model level. it seems these keys arent for all llm provider.